File: apps/android/shared/src/main/kotlin/za/co/naleli/z83/shared/Identity.kt
package za.co.naleli.z83.shared

import kotlinx.serialization.Serializable

@Serializable
data class User(
    val id: String,
    val email: String,
    val phone: String? = null,
    val role: UserRole,
    val fullName: String,
    val emailVerifiedAt: String? = null,
    val deletedAt: String? = null,
    val createdAt: String,
)

File: packages/types/src/identity.ts
import type { UserRole } from "./enums.js";

export interface User {
  id: string;
  email: string;
  phone: string | null;
  role: UserRole;
  fullName: string;
  emailVerifiedAt: string | null;
  deletedAt: string | null;
  createdAt: string;
}

export interface AdminUser {
  id: string;
  userId: string;
  permissionLevel: "verifier" | "superadmin";
}

export interface CafeAccount {
  id: string;
  name: string;
  province: string;
  address: string | null;
  contactEmail: string | null;
  contactPhone: string | null;
}

export interface CafeStaff {
  id: string;
  userId: string;
  cafeAccountId: string;
}

export interface AssistedSession {
  id: string;
  cafeStaffId: string;
  applicantUserId: string;
  status: "pending" | "open" | "closed";
  openedAt: string;
  authorizedAt: string | null;
  closedAt: string | null;
  openedReason: string | null;
}
